app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,54 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_amazon_host_instance(new_result)
+      _, az, uid = URI(new_result[:identity_infra]).path.split('/')
+      az_id = AvailabilityZone.find_by(:name => az).try(:id)
+      ManageIQ::Providers::Amazon::CloudManager::Vm.find_by(
+        :uid_ems              => uid,
+        :availability_zone_id => az_id,
+        :type                 => ManageIQ::Providers::Amazon::CloudManager::Vm.name)
+    end
+
+    def find_google_host_instance(new_result)
+      uri = URI(new_result[:identity_infra])
+      _, az, instance_name = uri.path.split('/')
+      az_id = AvailabilityZone.find_by(:name => az).try(:id)
+      ManageIQ::Providers::Google::CloudManager.find_by_project(uri.host).vms.find_by(
+        :name                 => instance_name,
+        :availability_zone_id => az_id)
+    end
+
+    def find_host_by_uuid(new_result)
+      uuid_provider_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                             ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                             ManageIQ::Providers::Vmware::InfraManager::Vm.name]
```

Comment:
@enoodle isn't the `ProviderID` giving you an insight on what would be the type? (w.g. `aws://`, `gce://`, etc.)